server/models/Cart.js

```diff
@@ -1,26 +1,40 @@
-const mongoose = require('mongoose')
+const mongoose = require('mongoose');
 const CartSchema = new mongoose.Schema({
-    totalItemsPrice: {
+  user: {
+    // *This connect the user to his Cart.
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'user',
+  },
+  items: [
+    // *This will hold all the items that the user add to his cart.
+    {
+      item: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'item',
+      },
+    },
+  ],
+  price: [
+    {
+      totalItemsPrice: {
```

Comment:
You don't calculate the total price on the DB. It's better to do it on the client-side. It's short and simple.

Reply:
The idea was save all the parameters in DB and calc it at the end (client side ).
I want save all the information before the customer pay to us.